karapace/backup/api.py

```diff
@@ -384,14 +391,17 @@ def _handle_producer_send(
         instruction.key,
         instruction.value,
     )
-    producer.send(
-        instruction.topic_name,
-        key=instruction.key,
-        value=instruction.value,
-        partition=instruction.partition_index,
-        headers=[(key.decode() if key is not None else None, value) for key, value in instruction.headers],
-        timestamp_ms=instruction.timestamp,
-    ).add_errback(_raise_backup_error)
+    try:
+        producer.send(
+            instruction.topic_name,
+            key=instruction.key,
+            value=instruction.value,
+            partition=instruction.partition_index,
+            headers=[(key.decode() if key is not None else None, value) for key, value in instruction.headers],
+            timestamp_ms=instruction.timestamp,
+        ).add_errback(_raise_backup_error)
```

Comment:
The error raised in the callback is generic backup error. Should it be also treated as data restoration error?

Reply:
Instead of this catch and re-raise, perhaps we could just change the exception in `_raise_backup_error`?

Reply:
`_raise_backup_error` now raises `BackupDataRestorationError` directly.